packages/block-editor/src/store/reducer.js

```diff
@@ -1469,6 +1469,27 @@ export function isSelectionEnabled( state = true, action ) {
 	return state;
 }
 
+/**
+ * Reducer returning whether to display a prompt when certain blocks are removed.
+ *
+ * @param {boolean} state  Current state.
+ * @param {Object}  action Dispatched action.
+ *
+ * @return {boolean} Updated state.
+ */
+export function isRemovalPromptDisplayed( state = false, action ) {
+	switch ( action.type ) {
+		case 'PROMPT_REMOVAL':
+			return {
+				displayPrompt: action.displayPrompt,
+				removalFunction: action.removalFunction,
+				blockName: action.blockName,
+			};
+	}
+
+	return state;
+}
```

Comment:
I believe this could come down to something like

```js
case 'PROMPT_FOR_BLOCK_REMOVAL':
  return action.blockIds;
```

with the corresponding name change.